server/src/main/java/org/elasticsearch/health/GetHealthAction.java

```diff
@@ -87,6 +92,28 @@ public XContentBuilder toXContent(XContentBuilder builder, ToXContent.Params par
             builder.endObject();
             return builder.endObject();
         }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) {
+                return true;
+            }
+            if (o == null || getClass() != o.getClass()) {
+                return false;
+            }
+            Response response = (Response) o;
+            return clusterName.equals(response.clusterName) && status == response.status && components.equals(response.components);
```

Comment:
Interestingly we use a `List<HealthComponentResult>` here and that's ok because `HealthComponentResult` is a `record` which has a list of `HealthIndicatorResult` which is also a `record` but that last one has a `HealthIndicatorDetails` member and I'm not sure how we can guarantee that the `equals` method does not use objects references?

Reply:
I guess, the best we could do now is to assert (using single assert against expected newly created object). This way to would fail when equals/hashCode are not defined. Unfortunately I do not see a better way to guarantee this compile time

Reply:
I wonder if we should change `HealthIndicatorDetails` to an abstract class with: 

```
// Force implementations to override equals for consistency with `HealthIndicatorResult` record
@Override
public abstract boolean equals(Object obj);
```